packages/http-client-python/generator/pygen/codegen/serializers/general_serializer.py

```diff
@@ -42,8 +48,73 @@ def serialize_setup_file(self) -> str:
         params.update({"options": self.code_model.options})
         return template.render(code_model=self.code_model, **params)
 
-    def serialize_package_file(self, template_name: str, **kwargs: Any) -> str:
+    def _extract_min_dependency(self, s):
+        # Extract the minimum version from a dependency string.
+        #
+        # Handles formats like:
+        # - >=1.2.3
+        # - >=0.1.0b1 (beta versions)
+        # - >=1.2.3rc2 (release candidates)
+        #
+        # Returns the parsed version if found, otherwise version "0".
+        m = re.search(r"[>=]=?([\d.]+(?:[a-z]+\d+)?)", s)
+        return parse_version(m.group(1)) if m else parse_version("0")
+
+    def _keep_pyproject_fields(self, file_path: "Path") -> dict:
+        # Load the pyproject.toml file if it exists and extract fields to keep.
+        result: dict = {"KEEP_FIELDS": {}}
+        try:
+            with open(file_path, "rb") as f:
+                loaded_pyproject_toml = tomllib.load(f)
+        except Exception:  # pylint: disable=broad-except
+            # If parsing the pyproject.toml fails, we assume the it does not exist or is incorrectly formatted.
+            return result
+
+        # Keep azure-sdk-build configuration
+        if "tool" in loaded_pyproject_toml and "azure-sdk-build" in loaded_pyproject_toml["tool"]:
+            result["KEEP_FIELDS"]["tool.azure-sdk-build"] = loaded_pyproject_toml["tool"]["azure-sdk-build"]
```

Comment:
I try to use this PR to regenerate for existing SDKs and find some content of pyproject.toml is deleted like https://github.com/Azure/azure-sdk-for-python/compare/main...msyyc:test-2025-07-22?expand=1

<img width="2028" height="879" alt="image" src="https://github.com/user-attachments/assets/54bd1eb3-01c3-456b-9850-6caefd8f6b23" />

It seems that the logic doesn't work.

Reply:
Thanks for checking @msyyc! I repro'd this locally. It looks like it's not picking up the correct sdk root path since it's also not deleting the setup.py. Still looking into it.

Reply:
@msyyc - The issue should be fixed now!

Reply:
Could you add a similar test case in https://github.com/Azure/autorest.python/pull/3131 so that we could cover this scenario? For example, you could add one existing pyproject.toml with additional info for authentication-union test case then add test in https://github.com/Azure/autorest.python/tree/2f594b37becb5be429708af73193dfc1062dac6a/packages/typespec-python/test/generic_mock_api_tests/unittests to check whether the legacy info exists after regeneration.

Reply:
Looks good to me. After fix CI of https://github.com/Azure/autorest.python/pull/3131, I think the PR is close to merge.

Reply:
Approved.